QModManager/DataStructures/SortedTreeNode.cs

```diff
@@ -0,0 +1,313 @@
+                }
+            }
+
+            return SortResults.SortAfter;
+        }
+
+        private static SortResults NextLevelCompareBefore(SortedTreeNode<IdType, DataType> entity, SortedTreeNode<IdType, DataType> other)
+        {
+            if (entity.LoadAfter != null)
+            {
+                SortResults subResult = CompareLoadOrder(entity.LoadAfter, other);
+
+                switch (subResult)
+                {
+                    case SortResults.SortAfter:
+                        other.Error = ErrorTypes.CircularLoadOrder;
+                        entity.ChainInCircularLoadOrder();
+                        return SortResults.CircularLoadOrder;
+                    case SortResults.NoSortPreference:
+                        return SortResults.SortBefore;
+                    default:
+                        return subResult;
+                }
+            }
+
+            return SortResults.SortBefore;
+        }
+
+        public SortResults Sort(SortedTreeNode<IdType, DataType> other, bool testing = false)
+        {
+            SortResults topLevelResult = CompareLoadOrder(this, other);
+
+            SortResults midLevelResult = SortResults.NoSortPreference;
+            switch (topLevelResult)
+            {
+                case SortResults.DuplicateId:
+                case SortResults.CircularDependency:
+                case SortResults.CircularLoadOrder:
+                    return topLevelResult;
+                case SortResults.NoSortPreference:
+
+                    if (this.LoadBefore != null && this.LoadAfter != null)
+                    {
+                        SortResults testAfterResult = SortAfter(other, true);
+                        SortResults testBeforeResult = SortBefore(other, true);
+
+                        if (testAfterResult > SortResults.NoSortPreference)
+                        {
+                            return testAfterResult;
+                        }
+
+                        if (testBeforeResult > SortResults.NoSortPreference)
+                        {
+                            return testBeforeResult;
+                        }
+
+                        midLevelResult = testAfterResult > testBeforeResult
+                            ? testAfterResult
+                            : testBeforeResult;
+                    }
+                    else if (this.LoadBefore == null && this.LoadAfter != null)
+                    {
+                        SortResults testAfterResult = SortAfter(other, true);
+
+                        if (testAfterResult > SortResults.NoSortPreference)
+                        {
+                            return testAfterResult;
+                        }
+
+                        midLevelResult = testAfterResult;
+                    }
+                    else if (this.LoadAfter == null && this.LoadBefore != null)
+                    {
+                        SortResults testBeforeResult = SortBefore(other, true);
```

Comment:
Replace literal bool values with named constants for clarity.

Reply:
Is it really necessary?

Reply:
I look at `SortBefore(other, true)` and it doesn't make sense what makes it different from `SortBefore(other)`.

I'd prefer it to be more readable, even if it's functionally identical.